lib/autoload.fish

```diff
@@ -10,18 +10,46 @@
 #   Returns != 0 if all paths are missing.
 
 function autoload -d "autoload a function or completion path"
-  for path in $argv
+  set -l paths $argv
+
+  switch "$argv[1]"
+  case '-e' '--erase'
+    set erase
+
+    if test (count $argv) -ge 2
+      # Unshift path to remove option
+      set paths $argv[2..-1]
+    else
+      # Option called without arguments
+      return 1
+    end
+  end
+
+  for path in $paths
+
```

Comment:
Using three lines of code for a `continue`: one blank line and another two for code.

As this is not the main feature, I'd convert the three lines into `not test -d $path`; and continue`.